### ISSUE TITLE
TypeError in doc_writer.py

### ISSUE THREAD

**Issue by Purvanshsingh** [HTTP-APIs/hydra-python-core#72]
### I'm submitting a 
- [x] bug report.
- [ ] feature request.

### Current Behaviour:

While using query  querying_mechanism.py "**TypeError: Cannot mix str and non-str arguments**" occurred in doc_writer.py.

![Screenshot from 2021-04-14 17-50-53](https://user-images.githubusercontent.com/49719371/114710766-b9c90880-9d4b-11eb-95e8-8d8643df7f65.png)

### Expected Behaviour:

querying_mechanism.py should work normally, the issue is due to doc_maker.py which is passing **dict** instead of **str** in API_NAME argument.

### Steps to reproduce:

python querying_mechanism.py
url>>>http://localhost:8080/serverapi/

### Do you want to work on this issue?

yes

**Comment by Mec-iS** [comment]
Have you checked that the `API` variable is set correctly?

**Comment by Purvanshsingh** [comment]
I have tested it with the default hydrus server,
and if we don't set the variables then the default else block should be executed.

i.e

![Screenshot from 2021-04-14 18-28-26](https://user-images.githubusercontent.com/49719371/114714040-58a33400-9d4f-11eb-8bbc-8a02887b5550.png)

The entrypoint argument is a **dict** object which is being received if the else block is executed, but it should the **str**.
I have checked it by printing both the arguments base_url & API.

![Screenshot from 2021-04-14 17-50-53](https://user-images.githubusercontent.com/49719371/114714294-9a33df00-9d4f-11eb-8303-f7c38137f3e7.jpg)

The solution Is to change arguments to `entrypoint['@id']`.
It worked.

![Screenshot from 2021-04-14 18-07-51](https://user-images.githubusercontent.com/49719371/114715066-52fa1e00-9d50-11eb-812e-1b67d8c904d0.png)
